### ISSUE TITLE
Support comments when enabling/disabling 

### ISSUE THREAD

**Issue by chavacava** [mgechev/revive#73]
**Is your feature request related to a problem? Please describe.**
Currently, we can enable/disable `revive` with comments like

```go
//revive:disable
//revive:disable-next-line:exported,random
```

But the enabling/disabling comment pattern does not allow to add a, sometimes useful, explanation of why the developer is enabling/disabling the linter. 

**Describe the solution you'd like**

It could be nice if we can write enabling/disabling directives like

```go
//revive:disable:cyclomatic - for the moment we accept this complex function
//revive:disable:deep-exit  as discussed with Thom, in this case, it is not bad to exit from here 
```

**Comment by mgechev** [comment]
A comment above seems sufficient to me:

```
// False negative
//revive:disable
```

**Comment by chavacava** [comment]
> A comment above seems sufficient to me:

That was my first impression when my colleagues ask for this feature.

One "problem" with a comment above the directive is that both are somewhat decoupled, thus it is possible, for example, to delete one without deleting the other. 
Moreover, if the directive has the `line` modifier, putting a comment in the precedent line does not facilitate code reading.


```go
func (p *exporter) marker(m byte) {
	p.rawByte(m)
	// Enable this for help tracking down the location
	// of an incorrect marker when running in debugFormat.
        // False positive
	if false && trace { // revive:disable-line:bool-literal-in-expr 
		p.tracef("#%d ", p.written)
	}
	p.rawInt64(int64(p.written))
}
```

**Comment by chavacava** [comment]
Closing. Requested feature will not be implemented.
